src/app/core/routing/routing.service.ts

```diff
@@ -0,0 +1,197 @@
+import { Injectable } from '@angular/core';
+import { Router, RoutesRecognized } from '@angular/router';
+import { filter } from 'rxjs/operators/filter';
+import { Store, select } from '@ngrx/store';
+import { selectRoutedPageEpisodes, selectRouter } from '../../ngrx/reducers/selectors/';
+import {
+  RouterParams, Episode,
+  MetricsType, ChartType, CHARTTYPE_PODCAST,
+  INTERVAL_DAILY,
+  METRICSTYPE_DEMOGRAPHICS,
+  METRICSTYPE_DOWNLOADS,
+  METRICSTYPE_TRAFFICSOURCES } from '../../ngrx/';
+import * as localStorageUtil from '../../shared/util/local-storage.util';
+import * as dateUtil from '../../shared/util/date/';
+import * as ACTIONS from '../../ngrx/actions/';
+
+@Injectable()
+export class RoutingService {
+  routerParams: RouterParams;
+  episodes: Episode[];
+
+  constructor (public store: Store<any>,
+               private router: Router) {
+    this.dontAllowRoot();
+    this.subRoutedPageEpisodes();
+    this.subAndCheckRouterParams();
+  }
+
+  dontAllowRoot() {
+    // for redirecting users routed to '/'
+    this.router.events.pipe(
+      filter(event => event instanceof RoutesRecognized)
+    ).subscribe((event: RoutesRecognized) => {
+      if (event.url === '/' && this.routerParams && this.routerParams.podcastId) {
+        this.normalizeAndRoute(this.routerParams);
+      }
+    });
+  }
+
+  subRoutedPageEpisodes() {
+    this.store.pipe(select(selectRoutedPageEpisodes)).subscribe((episodes: Episode[]) => {
+      this.episodes = episodes;
+    });
+  }
+
+  subAndCheckRouterParams() {
+    this.store.pipe(select(selectRouter)).subscribe(routerParams => {
```

Comment:
this service watches the router params state and a) makes sure the app is loading with viable or default router params and b) loads the data we need when route changes